src/main/kotlin/no/nav/familie/ef/sak/journalføring/JournalpostClient.kt

```diff
@@ -40,29 +44,42 @@ class JournalpostClient(
     private val dokarkivUri: URI = integrasjonerConfig.dokarkivUri
 
     fun finnJournalposter(journalposterForBrukerRequest: JournalposterForBrukerRequest): List<Journalpost> {
+        validerIkkeUtviklerMedVeilederrolle()
         return postForEntity<Ressurs<List<Journalpost>>>(journalpostURI, journalposterForBrukerRequest).data
             ?: error("Kunne ikke hente vedlegg for ${journalposterForBrukerRequest.brukerId.id}")
     }
 
     fun finnJournalposterForBrukerOgTema(journalposterForBrukerOgTemaRequest: JournalposterForVedleggRequest): List<Journalpost> {
+        validerIkkeUtviklerMedVeilederrolle()
         return postForEntity<Ressurs<List<Journalpost>>>(URI.create("$journalpostURI/temaer"), journalposterForBrukerOgTemaRequest).data
             ?: error("Kunne ikke hente vedlegg for ${journalposterForBrukerOgTemaRequest.brukerId.id}")
     }
 
     fun hentJournalpost(journalpostId: String): Journalpost {
+        validerIkkeUtviklerMedVeilederrolle()
         val ressurs = try {
             getForEntity<Ressurs<Journalpost>>(URI.create("$journalpostURI?journalpostId=$journalpostId"))
         } catch (e: RessursException) {
             if (e.message?.contains("Fant ikke journalpost i fagarkivet") == true) {
-                throw ApiFeil("Finner ikke journalpost i fagarkivet", HttpStatus.BAD_REQUEST)
+                throw ApiFeil("Finner ikke journalpost i fagarkivet", BAD_REQUEST)
             } else {
                 throw e
             }
         }
         return ressurs.getDataOrThrow()
     }
 
+    private fun validerIkkeUtviklerMedVeilederrolle() {
+        if (featureToggleService.isEnabled(Toggle.UTVIKLER_MED_VEILEDERRROLLE)) {
+            throw ApiFeil(
+                "Kan ikke hente ut journalposter som utvikler med veilederrolle. Kontakt teamet dersom du har saksbehandlerrolle.",
+                FORBIDDEN,
+            )
+        }
+    }
```

Comment:
```suggestion
    private fun kastApiFeilDersomUtviklerMedVeilederrolle() {
        if (featureToggleService.isEnabled(Toggle.UTVIKLER_MED_VEILEDERRROLLE)) {
            throw ApiFeil(
                "Kan ikke hente ut journalposter som utvikler med veilederrolle. Kontakt teamet dersom du har saksbehandlerrolle.",
                FORBIDDEN,
            )
        }
    }
```

Navnet på denne metoden ble litt forvirrende for meg. Jeg skjønner at man kun skal gå videre dersom man ikke er utvikler med veilederrolle, men metoden sjekker om man er utvikler med veilerrolle og thrower i såfall exception 🤔 

For meg ville en rename til `kastApiFeilDersomUtviklerMedVeilederrolle()` vært mer presis :)

Reply:
Gode argumenter og finfine innspill, @ViktorGSolberg! 
I'll fix 👷